File: app/src/main/java/com/reactive/owm/domain/server/ServerGateway.java
package com.reactive.owm.domain.server;

import com.reactive.owm.entities.ForecastsResponse;

import io.reactivex.Single;
import retrofit2.http.GET;
import retrofit2.http.Query;

public interface ServerGateway {

    @GET("/data/2.5/forecast")
    Single<ForecastsResponse> requestFiveDaysForecasts(@Query("id") Long id);


}

File: kotlin-app/src/main/java/com/ahmedismail/app/domain/server/ServerGateway.kt
package com.ahmedismail.app.domain.server

import com.ahmedismail.app.entities.ForecastsResponse
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface ServerGateway {
    @GET("/data/2.5/forecast")
    fun requestFiveDaysForecasts(@Query("id") id: Long?): Single<ForecastsResponse>
}